odoo/fields.py

```diff
@@ -660,6 +660,47 @@ def get(key, get_context=env.context.get):
 
         return tuple(get(key) for key in self.depends_context)
 
+    def mapped(self, records):
+        """
+        Fetch self on all of the provided records
+
+        :param records: the records for which to fetch the current field (self)
+        :type records: recordset
+        :return: an ordered collection of the value of self for every record in records
+        :rtype: list or recordset
+
+        This method is meant to be used internally and has very little benefit over a simple call
+        to `~odoo.models.BaseModel.mapped()` on a recordset.
+        """
+        if self.name == 'id':
+            # not stored in cache
+            return list(records._ids)
+
+        if self.compute:
+            # a recompute must be forced for records to_recompute that intersect with input records
+            # otherwise an infinite loop will occur on the next call to flush() / recompute()
+            # as the value will be False in cache and will never trigger a MissingError which,
+            # in turn, triggers a recompute
+            to_compute_ids = records.env.all.tocompute.get(self, ())
+            ids = [id_ for id_ in records._ids if id_ in to_compute_ids]
+
+            if self.recursive:
+                # recursive computed fields are a special case and must be computed one by one
+                # in order, as a record down the stack may depend on the computed value of a
+                # previous record
+                ids = ids[:1]
```

Comment:
This is not correct: you have to force the computation of `self` on *all* records to compute!
```python
for record in records.browse(ids):
    self.__get__(record, type(records))
```
Note that this `for` loop works for all computed fields...  Maybe worth trying its performance.